wire-schema/src/main/java/com/squareup/wire/schema/PruningRules.kt

```diff
@@ -208,20 +210,20 @@ class PruningRules private constructor(builder: Builder) {
      * The exclusive lower bound of the version range. Fields with `until` values greater than this
      * are retained.
      */
-    fun oldest(oldest: Long?) = apply {
-      this.oldest = oldest
+    fun oldest(oldest: String?) = apply {
+      this.oldest = oldest?.toLowerCaseSemVer()
     }
 
     /**
      * The inclusive upper bound of the version range. Fields with `since` values less than or equal
      * to this are retained.
      */
-    fun newest(newest: Long?) = apply {
-      this.newest = newest
+    fun newest(newest: String?) = apply {
+      this.newest = newest?.toLowerCaseSemVer()
     }
 
     fun build(): PruningRules {
-      check((oldest ?: Long.MIN_VALUE) <= (newest ?: Long.MAX_VALUE)) {
+      check(oldest == null || newest == null || oldest!! <= newest!!) {
```

Comment:
worth splitting into 3 separate checks for debugging purposes?

Reply:
Unfortunately I only have the nulls there to guard the comparison.